api/src/APIHTTPService.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <rapidjson/document.h>
+#include <memory>
+
+#include "DB.h"
+#include "HTTPRequest.h"
+#include "HTTPRouter.h"
+
+namespace rustla2 {
+
+class APIHTTPService {
+ public:
+  explicit APIHTTPService(std::shared_ptr<DB> db);
+
+  void RegisterRoutes(HTTPRouter &router);
```

Comment:
I think this should this be a `const` reference?

Reply:
you're right, this is inconsistent - these should take pointers.